horace_core/sqw/@axes_block/private/realign_bin_edges_.m

```diff
@@ -0,0 +1,130 @@
+function ax_block_al = realign_bin_edges_(obj,ax_block)
+% align input axes block to have the same or commensurate
+% bin sizes as this axes block and the integration ranges equal
+% or smaller than the ranges of this axes block but
+% commensurate with this lattice
+%
+hc = hor_config;
+log_level = hc.log_level;
+want_pax = false(1,4);
+want_pax(ax_block.pax) = true;
+this_is_pax = false(1,4);
+this_is_pax(obj.pax) = true;
+
+this_range = obj.img_range;
+ax_range   = ax_block.img_range;
+ax_nbins   = ax_block.nbins_all_dims;
+this_nbins = obj.nbins_all_dims;
+for i=1:4
+    if want_pax(i) % pax requested
+        if this_is_pax(i)
+            [ax_nbins(i),ax_range(:,i)] = ...
+                realign_pax(i,this_range(:,i),this_nbins(i),ax_range(:,i),ax_nbins(i),log_level);
+        else % but was iax in this direction, so only iax possible
+            ax_nbins(i) = 1;
+            ax_range(:,i) = this_range(:,i);
+            if log_level>0
+                warning('HORACE:realign_bin_edges:invalid_argument', ...
+                    ['projection axis is requested in direction N%d, ',...
+                    'but original axis there is integration. ',...
+                    'Can not rebin integraion axis on dnd object.', ...
+                    ' Doing integration axis in this direction'],...
+                    i)
+            end
+        end
+    else  %iax requested
+        ax_nbins(i) = 1;
+        ax_range(:,i) = find_proper_iax_range(this_range(:,i),this_nbins(i),ax_range(:,i));
+    end
+end
+ax_block_al = axes_block(obj);
+ax_block_al.img_range=ax_range;
+ax_block_al.nbins_all_dims = ax_nbins;
+
+function iax_requested = find_proper_iax_range(in_range,nbins,iax_requested)
+% find integration range, which covers minimal axes range
+
+if iax_requested(1)<in_range(1)
+    iax_requested(1) = in_range(1);
+end
```

Comment:
```suggestion
iax_requested(1) = max(iax_requested(1), in_range(1));
```

Reply:
possible bit probably not worth spending time on modifying this